src/test/java/org/reaktivity/specification/kafka/internal/KafkaFunctionsTest.java

```diff
@@ -281,23 +277,109 @@ public void shouldGenerateMergedBeginExtensionWithAgeFilter()
+                                         .partition(0, 1L)
+                                         .filter()
+                                             .keyNot("match")
+                                             .build()
+                                         .filter()
+                                             .header("name", "value")
+                                             .build()
+                                         .build()
+                                     .build();
+
+        DirectBuffer buffer = new UnsafeBuffer(build);
+        KafkaBeginExFW beginEx = new KafkaBeginExFW().wrap(buffer, 0, buffer.capacity());
+        assertEquals(0x01, beginEx.typeId());
+        assertEquals(KafkaApi.MERGED.value(), beginEx.kind());
+
+        final KafkaMergedBeginExFW mergedBeginEx = beginEx.merged();
+        assertEquals("topic", mergedBeginEx.topic().asString());
+
+        assertNotNull(mergedBeginEx.partitions()
+                                   .matchFirst(p -> p.partitionId() == 0 && p.partitionOffset() == 1L));
+
+        final MutableInteger filterCount = new MutableInteger();
+        mergedBeginEx.filters().forEach(f -> filterCount.value++);
+        assertEquals(2, filterCount.value);
+        assertNotNull(mergedBeginEx.filters()
+                .matchFirst(f -> f.conditions()
+                .matchFirst(c -> c.kind() == NOT.value() && c.not().condition().kind() == KEY.value() &&
+                    "match".equals(c.not().condition().key()
+                                    .value()
+                                    .get((b, o, m) -> b.getStringWithoutLengthUtf8(o, m - o)))) != null));
         assertNotNull(mergedBeginEx.filters()
                 .matchFirst(f -> f.conditions()
                 .matchFirst(c -> c.kind() == HEADER.value() &&
                     "name".equals(c.header().name()
                                    .get((b, o, m) -> b.getStringWithoutLengthUtf8(o, m - o))) &&
                     "value".equals(c.header().value()
                                     .get((b, o, m) -> b.getStringWithoutLengthUtf8(o, m - o)))) != null));
+    }
+
+    @Test
+    public void shouldGenerateMergedBeginExtensionWithNullKeyAndHeaderNotEqualsFilter()
+    {
+        byte[] build = KafkaFunctions.beginEx()
+                                     .typeId(0x01)
+                                     .merged()
+                                         .topic("topic")
+                                         .partition(0, 1L)
+                                         .filter()
+                                             .keyNot(null)
+                                             .build()
+                                         .filter()
+                                             .headerNot("name", null)
+                                             .build()
+                                         .build()
+                                     .build();
```

Comment:
Is this intended to be `AND` or `OR` logic to combine the filter conditions?
It looks like the method name implies `AND` but the behavior implies `OR`.